diff --git a/dist/esm/internal/options.js b/dist/esm/internal/options.js
index f054963565e996ffe1225398148fe80296022480..54706d9ef24aa87cac100941482bd9b0dfd64dbc 100644
--- a/dist/esm/internal/options.js
+++ b/dist/esm/internal/options.js
@@ -58,7 +58,9 @@ export const buildOptions = ({ cliOptions }) => {
         maxThreads: process.env.PONDER_MAX_THREADS !== undefined
             ? Number(process.env.PONDER_MAX_THREADS)
             : 4,
-        factoryAddressCountThreshold: 1000,
+        factoryAddressCountThreshold: process.env.PONDER_FACTORY_ADDRESS_THRESHOLD !== undefined
+            ? Number(process.env.PONDER_FACTORY_ADDRESS_THRESHOLD)
+            : 1000,
         rpcMaxConcurrency: 256,
         // v8.getHeapStatistics().heap_size_limit / 5, rounded up to the nearest 64 MB
         indexingCacheMaxBytes: process.env.PONDER_CACHE_BYTES !== undefined
